### ISSUE TITLE
TTF: use-after-poison in OpenType::Cmap::subtable

### ISSUE THREAD

**Issue by hghwng** [SerenityOS/serenity#18304]
When fuzzing FuzzTTF, we found that the program crashed with use-after-poison, which indicates that the parser is reading past the allocated input buffer. Here's the report from Address Sanitizer:

```
==3717023==ERROR: AddressSanitizer: use-after-poison on address 0x1555526f6deb at pc 0x55555571a8cf bp 0x7fffffffd500 sp 0x7fffffffd4f8
READ of size 1 at 0x1555526f6deb thread T0
    #0 0x55555571a8ce in OpenType::be_u32(unsigned char const*) /root/build/serenity/Userland/Libraries/LibGfx/Font/OpenType/Font.cpp:39
    #1 0x5555557a6c7f in OpenType::Cmap::subtable(unsigned int) const /root/build/serenity/Userland/Libraries/LibGfx/Font/OpenType/Cmap.
cpp:71
    #2 0x555555723b1e in OpenType::Font::try_load_from_offset(AK::Span<unsigned char const>, unsigned int) /root/build/serenity/Userland
/Libraries/LibGfx/Font/OpenType/Font.cpp:526
    #3 0x555555721208 in OpenType::Font::try_load_from_externally_owned_memory(AK::Span<unsigned char const>, unsigned int) /root/build/
serenity/Userland/Libraries/LibGfx/Font/OpenType/Font.cpp:?
    #4 0x55555571a3c7 in LLVMFuzzerTestOneInput /root/build/serenity/Meta/Lagom/Fuzzers/FuzzTTF.cpp:13

SUMMARY: AddressSanitizer: use-after-poison /root/build/serenity/Userland/Libraries/LibGfx/Font/OpenType/Font.cpp:39 in OpenType::be_u32
(unsigned char const*)
Shadow bytes around the buggy address:
  0x02ab2a4d6d60: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x02ab2a4d6d70: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x02ab2a4d6d80: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x02ab2a4d6d90: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x02ab2a4d6da0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
=>0x02ab2a4d6db0: 00 00 00 00 00 00 00 00 00 00 00 00 00[03]f7 f7
  0x02ab2a4d6dc0: f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7
  0x02ab2a4d6dd0: f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7
  0x02ab2a4d6de0: f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7
  0x02ab2a4d6df0: f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7
  0x02ab2a4d6e00: f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7 f7
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
```

You can reproduce the bug with the following input (base64 encoded):

```
AAEAAAAKAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGNtYXAAAAAAAAABmAAAACwAAAAAAAAAAAAAAAAA
AAAAaGVhZAAAAAAAAAAAAAAANmhoZWEAAAAAAAAAAAAAACRobXR4AAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAG1heHAAAAAAAAAACAAAACBuYW1lAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
```

**Comment by navadev** [comment]
One thing that could be done in `Cmap::subtable` is to replace the usage of `m_slice.offset_pointer` with `m_slice.offset(` as the latter has boundary checks to ensure we don't go past the end of the buffer.

**Comment by tcl3** [comment]
I can't reproduce this. Likely fixed by e1b438bb and cdfe893
